src/sentry/api/endpoints/organization_details.py

```diff
@@ -380,6 +398,20 @@ def save(self):
             org.name = data["name"]
         if "slug" in data:
             org.slug = data["slug"]
+        if "providerName" in data and "providerConfig" in data:
+            provider_name = data["providerName"]
+            provider_config = data["providerConfig"]
+            with transaction.atomic():
+                auth_provider = AuthProvider.objects.update_or_create(
+                    organization_id=org.id,
+                    defaults={"provider": provider_name},
+                )[0]
+                provider = auth_provider.get_provider()
+                try:
+                    config = provider.build_config(provider_config)
```

Comment:
Can we add a `sentry-source: "api-organization-details"` to the config so we can track which configs are setup through this API vs other means?

If it turns out that this config can be abused (e.g. using a bad config in our AuthPipeline flow to initiate requests to other servers), we have a finite quantity of rows to moderate.

Reply:
Added and also modified tests to check for this